app/gui/src/dashboard/hooks/backendHooks.tsx

```diff
@@ -0,0 +1,1136 @@
+      const projectName = (() => {
+        const prefix = `${templateName ?? 'New Project'} `
+        const projectNameTemplate = new RegExp(`^${prefix}(?<projectIndex>\\d+)$`)
+        const projectIndices = siblings
+          .filter(backendModule.assetIsProject)
+          .map((item) => projectNameTemplate.exec(item.title)?.groups?.projectIndex)
+          .map((maybeIndex) => (maybeIndex != null ? parseInt(maybeIndex, 10) : 0))
+        return `${prefix}${Math.max(0, ...projectIndices) + 1}`
+      })()
+      const dummyId = backendModule.ProjectId(uniqueString())
+      const path = backend instanceof LocalBackend ? backend.joinPath(parentId, projectName) : null
+
+      const placeholderItem: backendModule.ProjectAsset = {
+        type: AssetType.project,
+        id: dummyId,
+        title: projectName,
+        modifiedAt: toRfc3339(new Date()),
+        parentId,
+        permissions: tryCreateOwnerPermission(
+          `${parentPath ?? ''}/${projectName}`,
+          category,
+          user,
+          users ?? [],
+          userGroups ?? [],
+        ),
+        projectState: {
+          type: backendModule.ProjectState.placeholder,
+          volumeId: '',
+          openedBy: user.email,
+          ...(path != null ? { path } : {}),
+        },
+        labels: [],
+        description: null,
+      }
+
+      insertAssets([placeholderItem], parentId)
+
+      return await createProjectMutation
+        .mutateAsync([
+          {
+            parentDirectoryId: placeholderItem.parentId,
+            projectName: placeholderItem.title,
+            ...(templateId == null ? {} : { projectTemplateName: templateId }),
+            ...(datalinkId == null ? {} : { datalinkId: datalinkId }),
+          },
+        ])
+        .catch((error) => {
+          void deleteAsset(placeholderItem.id, parentId)
+          toastAndLog('createProjectError', error)
+          throw error
+        })
+        .then((createdProject) => {
+          doOpenProject({
+            id: createdProject.projectId,
+            type: backend.type,
+            parentId: placeholderItem.parentId,
+            title: placeholderItem.title,
+          })
+          return createdProject
+        })
+    },
+  )
+}
+
+/** A function to create a new secret. */
+export function useNewSecret(backend: Backend, category: Category) {
+  const insertAssets = useInsertAssets(backend, category)
+  const toggleDirectoryExpansion = useToggleDirectoryExpansion()
+  const { user } = useFullUserSession()
+  const { data: users } = useBackendQuery(backend, 'listUsers', [])
+  const { data: userGroups } = useBackendQuery(backend, 'listUserGroups', [])
+  const createSecretMutation = useMutation(backendMutationOptions(backend, 'createSecret'))
+
+  return useEventCallback(
+    async (
+      name: string,
+      value: string,
+      parentId: DirectoryId,
+      parentPath: string | null | undefined,
+    ) => {
```

Comment:
not sure if these shouldbe `useMutation` instead of `useEventCallback`. @MrFlashAccount wdyt?